Trigger "added" and "removed" events when adding children on Container

endel opened the pull request:
Hello there,

This is similar to Flash's `Event.ADDED_TO_STAGE` behavior.
I've added a simple test for that. 

**Usage example:**

```javascript
class Something extends PIXI.Container {

  constructor () 
  {
    super();

    this.on('added', this.onAdded, this);
    this.on('removed', this.onRemoved, this);
  }

  onAdded (to) 
  {
    console.log("I'm now attached to some Container!", to)
  }

  onRemoved (from) 
  {
    console.log("I was removed from the Container. :(", from)
  }

}
```

englercj commented:
I like the idea of an event happening when a child is added!

I don't like the change to the parent property. This property is used a few times in rendering paths, so making it a getter will likely cause a perf hit. Does it not make sense to emit this event in the addChild methods?

endel commented:
@englercj Thanks for your feedback!

I've made the changes you suggested and squashed the commits into a single one.

Now it makes more sense, since when the event is triggered it's possible to access container's children in a consistent state. I've also added a test for that.

englercj commented:
Thanks! I also suggest sending the container that it was added/remove to/from as the param of the event, so people can access that if they need it.

endel commented:
Nice! I've added it as the param. Also updated the test. Sounds good for you? :)

englercj commented:
Looks great to me! If @GoodBoyDigital is OK with it, lets merge it in.

GoodBoyDigital commented:
Will give this a review over the weekend chaps. Good stuff

On Saturday, 6 June 2015, Chad Engler <notifications@github.com> wrote:

> Looks great to me! If @GoodBoyDigital <https://github.com/GoodBoyDigital>
> is OK with it, lets merge it in.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoodBoyDigital/pixi.js/pull/1870#issuecomment-109534490>
> .
>


-- 
Mat Groves

*Technical Partner*


 Telephone: 07708 114496 :: www.goodboydigital.com
 First Floor, Unit 9B, Queens Yard, White Post Lane, London, E9 5EN
 goodboy©. All Rights Reserved.

GoodBoyDigital commented:
looking good. This could turn out to be quite handy! thanks @endel :+1:
